### ISSUE TITLE
[Feature-Request] Incorporation of `tflite_flutter_helper` for Image Classification Tasks

### ISSUE THREAD

**Issue by saurabhkumar8112** [tensorflow/flutter-tflite#171]
As many of you might already be familiar, `tflite_flutter_helper` is a popular helper library, specifically for Image processing while dealing with tflite. This was earlier developer by tensorflow team as [well](https://pub.dev/packages/tflite_flutter_helper)

As of now it has been [discontinued,](https://pub.dev/packages/tflite_flutter_helper) but it has good helper functions for tensor operations, which are quite efficient.

For example, using experiments I found that using `tflite_helper_liibrary` to run mobilenet inference on Android GPU Delegate, its **8X faster** compared to simply using Image and then converting it into ListMatrix(as suggested in the mobilenet example).

If that's the case its important to use `tflite_flutter_helper` with `tflite_flutter`. It would be nice this package can either be incorporated or taken up by Tensorflow team as well(given that they have bandwidth). Would be nice to have

How to verify the inference speed
1. Use a XNNPackDelegate for android.
2. Run inference using TensorImage Wrapper class.
3. Do step 2 but with vanilla Image inference(as given in mobilenet examples)

You will see that doing inference in step 2 is around 8X faster than step 3. I did this evaluation in Pixel 3. 
PS: This doesn't include any pre-processing time benchmarks, just inferencing.

**Comment by PaulTR** [comment]
So it actually wasn't written by the TFLite team - it was written by a Summer of Code participant a few years ago :) Unfortunately bandwidth is _very_ limited (no one other than myself from the TFLite team is actually working on this repo, so it's very community driven and just maintained/reviewed). Rather than the helper library though, the Flutter team is actively working on the [MediaPipe](https://developers.google.com/mediapipe) plugin https://github.com/google/flutter-mediapipe/pulls which should address the speed/ease issues when it's ready.

**Comment by JobiJoba** [comment]
@saurabhkumar8112 Could you put an example where you use that library? 

The original library for me doesn't work with some phone which are supported by tensorflow and I think it come from the image processing part (doesn't transform correctly some format or else) as it is detedcting stuff with a very bad confidence...

**Comment by JobiJoba** [comment]
Oh wow, Thank you for your issue :P 
Make me look into the convert image of the Live Object detection and see that the problem was coming from there. (I change to another way of converting and now it works on my Oppo Reno 7Z).

**Comment by saurabhkumar8112** [comment]
@JobiJoba were you able to resolve the issue or still want help?

**Comment by saurabhkumar8112** [comment]
Hey @PaulTR thanks for replying. I look forward to MediaPipe.

**Comment by JobiJoba** [comment]
kinda solve but it's slow :) but at least I detect something :)

**Comment by saurabhkumar8112** [comment]
yes @JobiJoba object detection is likely to be slow. Make sure you're using the required GPU delegate

**Comment by JobiJoba** [comment]
It seems unfortunately :( 

```
static Future<Interpreter> _loadModel() async {
    final interpreterOptions = InterpreterOptions();

    // Use XNNPACK Delegate
    if (Platform.isAndroid) {
      interpreterOptions.addDelegate(XNNPackDelegate());
    }

    return Interpreter.fromAsset(
      _modelPath,
      options: interpreterOptions..threads = 4,
    );
  }
```

I try to make the "example" project of live object detection working and see what kind of FPS I can achieve to then continue on something more complex. 

On Android, only XNNPackDelegate is working ; GPUDelegate ; GpuDelegateV2 is not working. 

Did I do something wrong or they are suppose to work?

**Comment by saurabhkumar8112** [comment]
Try reducing/increasing the number of threads. Depending upon device, that might increase performance. There's always a threadoff between spawing threads, managing them and the extra processing power they can provide.

**Comment by saurabhkumar8112** [comment]
depends on the model as well. SSD produces 8k bounding boxes, and Retinanet produces ~100k

**Comment by andynewman10** [comment]
> So it actually wasn't written by the TFLite team - it was written by a Summer of Code participant a few years ago :) 

You did advertise the Flutter package availability with bells and whistles - remember the following post? It's from August 2023, not so long ago:

https://blog.tensorflow.org/2023/08/the-tensorflow-lite-plugin-for-flutter-officially-available.html

Please understand that, when developers see such a blog post, they understand that Google is taking some form of ownership of TFLite for Flutter, through this package. Which isn't true, to say the least.

> Unfortunately bandwidth is _very_ limited (no one other than myself from the TFLite team is actually working on this repo, so it's very community driven and just maintained/reviewed). Rather than the helper library though, the Flutter team is actively working on the [MediaPipe](https://developers.google.com/mediapipe) plugin https://github.com/google/flutter-mediapipe/pulls which should address the speed/ease issues when it's ready.

Based on your various answers in this repository, I understand that developers are strongly encouraged to give up high hopes on `tflite_flutter` and keep an eye on [MediaPipe](https://developers.google.com/mediapipe).

**Comment by PaulTR** [comment]
Remember, people are working on these things because they think they're valuable and are trying to make things easier for others. Please try and keep feedback constructive.

> Please understand that, when developers see such a blog post, they understand that Google is taking some form of ownership of TFLite for Flutter, through this package. Which isn't true, to say the least.

The post says it was migrated to a place where it can be maintained (which it is, there are actively PRs being merged and responses). Before this the TFLite plugin hadn't been updated in three years.

> Based on your various answers in this repository, I understand that developers are strongly encouraged to give up high hopes on tflite_flutter and keep an eye on [MediaPipe](https://developers.google.com/mediapipe).

The plugins are for fairly different things. TensorFlow Lite, on all platforms, is very much for custom down-in-the-weeds solutions, and other tools have been made for common tasks like image classification. We're moving in the same direction for Flutter, so right now it's just an in-between time now that I've drummed up people to work on the MediaPipe plugin compared to having nothing at all.

**Comment by saurabhkumar8112** [comment]
Hey @PaulTR any update on this? Is it added in a potential todo? Would be a huge support

**Comment by saurabhkumar8112** [comment]
Hey @PaulTR I see that mediaPipe 0.0.1 was released but it only has helpers for text-gen/text-embedding models like BERT/GPT.
The media-pipe vision is still empty. I wanted to put a reminder here regarding incorporating tflite_flutter_helper in MediaPipe vision.

**Comment by PaulTR** [comment]
I believe it's on the roadmap :) Craig over with the Flutter team is
leading that

**Comment by AntoineChauviere** [comment]
Do you know how to perform image processing with MediaPipe in Flutter? My code is based on tflite_flutter_helper, and it doesn’t seem to work anymore.
